celer/tests/test_mtl.py

```diff
@@ -230,23 +228,17 @@ def test_weights_group_lasso():
 
     assert_allclose(alphas1, alphas2)
     assert_allclose(
-        coefs1, coefs2 / augmented_weights[:, None], atol=1e-4, rtol=1e-3)
+        coefs1, coefs2 / augmented_weights[:, None], rtol=1e-3)
     assert_array_less(gaps1, tol * norm(y) ** 2 / len(y))
     assert_array_less(gaps2, tol * norm(y) ** 2 / len(y))
 
-    # fit
-    alpha = 0.001
-    clf1 = GroupLasso(groups, alpha=alpha, weights=weights,
-                      fit_intercept=False).fit(X, y)
-    clf2 = GroupLasso(groups, alpha=alpha, fit_intercept=False).fit(
-        X.multiply(1. / augmented_weights), y)
-
-    assert_allclose(clf1.coef_, clf2.coef_ / augmented_weights)
 
-    # weights must be > 0
-    clf1.weights[0] = 0.
-    np.testing.assert_raises(ValueError, clf1.fit, X=X, y=y)
-    return
+def test_check_weights():
```

Comment:
To test whether GroupLasso raises an error if any of the weights is null. Isn't it?

Reply:
I don't get your question

Reply:
What is the purpose of this test?

Reply:
Ah, ok ! Yes, it is the purpose of this test.
Each test should test a précisé aspect of the code, so i isolated it from the solver test

Reply:
Does it make the test more robust if the null weight is set randomly?

Reply:
Maybe marginally. It's not worth doing it.